### ISSUE TITLE
[🐞] Cloudflare Pages integration is incompatitble with Supabase.

### ISSUE THREAD

**Issue by niwatoro** [BuilderIO/qwik#5039]
### Which component is affected?

Qwik Rollup / Vite plugin

### Describe the bug

I aimed to host a web app that connects with Supabase on Cloudflare Pages. To achieve this, I followed a series of steps:

1. Initially, I utilized the following commands:

```
npm create qwik@latest
cd qwik-app
npm run qwik add cloudflare-pages
npm install supabase-auth-helpers-qwik
```

2. Subsequently, I integrated Supabase functionality into the app by adding a script in `src/routes/index.tsx`. This script was based on an example I found in the [documentation](https://qwik.builder.io/docs/integrations/supabase/).

```typescript
import { createServerClient } from "supabase-auth-helpers-qwik";
import { routeLoader$, component$ } from "@qwik/server";

export const useItems = routeLoader$(async (requestEv) => {
  const supabase = createServerClient(
    requestEv.env.get("VITE_SUPABASE_URL")!,
    requestEv.env.get("VITE_SUPABASE_ANON_KEY")!,
    requestEv
  );
  const { data: items } = await supabase.from("items").select("*");
  return items ?? [];
});

export default component$(() => {
  const items = useItems();

  return <div>You have {items.value.length} Items!</div>;
});
```

3. However, this led to a build error. 

```
[commonjs--resolver] Cannot bundle Node.js built-in "stream" imported from "node_modules/@supabase/node-fetch/lib/index.mjs". Consider disabling ssr.noExternal or remove the built-in dependency.
error during build:
RollupError: Cannot bundle Node.js built-in "stream" imported from "node_modules/@supabase/node-fetch/lib/index.mjs". Consider disabling ssr.noExternal or remove the built-in dependency.
    at error (qwik-supabase-bug-reproduce/node_modules/rollup/dist/es/shared/node-entry.js:2287:30)
    at Object.error (qwik-supabase-bug-reproduce/node_modules/rollup/dist/es/shared/node-entry.js:25268:20)
    at Object.resolveId (qwik-supabase-bug-reproduce/node_modules/vite/dist/node/chunks/dep-3b8eb186.js:28210:34)
    at Object.handler (qwik-supabase-bug-reproduce/node_modules/vite/dist/node/chunks/dep-3b8eb186.js:48180:19)
    at qwik-supabase-bug-reproduce/node_modules/rollup/dist/es/shared/node-entry.js:25461:40
    at async PluginDriver.hookFirstAndGetPlugin (qwik-supabase-bug-reproduce/node_modules/rollup/dist/es/shared/node-entry.js:25361:28)
    at async resolveId (qwik-supabase-bug-reproduce/node_modules/rollup/dist/es/shared/node-entry.js:24035:26)
    at async ModuleLoader.resolveId (qwik-supabase-bug-reproduce/node_modules/rollup/dist/es/shared/node-entry.js:24449:15)
    at async Object.resolveId (qwik-supabase-bug-reproduce/node_modules/vite/dist/node/chunks/dep-3b8eb186.js:7984:10)
    at async PluginDriver.hookFirstAndGetPlugin (qwik-supabase-bug-reproduce/node_modules/rollup/dist/es/shared/node-entry.js:25361:28)
```

4. It seemed that the code was attempting to use Node.js built-in APIs within the browser. In order to address this issue, I substituted these APIs with browser-compatible alternatives. This did resolve the error, although the resulting output files were noticeably larger. I held concerns that this solution might compromise Qwik's otherwise impressive performance.

If you have any recommendations or insights on how to maintain Qwik's performance while addressing the build error, I'd greatly appreciate your guidance.

### Reproduction

https://github.com/niwatoro/qwik-supabase-pages-bug-reproduce

### Steps to reproduce

 1. Clone this repository.
 2. Run `npm run build`.

### System Info

```shell
System:
    OS: macOS 13.0
    CPU: (8) arm64 Apple M1
    Memory: 72.36 MB / 8.00 GB
    Shell: 5.8.1 - /bin/zsh
  Binaries:
    Node: 20.5.0 - ~/.nvm/versions/node/v20.5.0/bin/node
    Yarn: 1.22.19 - ~/.nvm/versions/node/v20.5.0/bin/yarn
    npm: 9.8.1 - ~/.nvm/versions/node/v20.5.0/bin/npm
    pnpm: 8.7.0 - ~/.nvm/versions/node/v20.5.0/bin/pnpm
  Browsers:
    Chrome: 116.0.5845.110
    Safari: 16.1
  npmPackages:
    @builder.io/qwik: ^1.2.10 => 1.2.10 
    @builder.io/qwik-city: ^1.2.10 => 1.2.10 
    undici: 5.23.0 => 5.23.0 
    vite: 4.4.9 => 4.4.9
```

## Tentative Fix

1. Install the following browserify packages:

```bash
npm install assert buffer events stream-http https-browserify punycode stream-browserify url util browserify-zlib
```

2. Add the following to `vite.config.ts`:

```typescript
export default defineConfig(() => {
  return {
    resolve: {
      alias: {
        assert: "assert",
        buffer: "buffer",
        events: "events",
        http: "stream-http",
        https: "https-browserify",
        punycode: "punycode",
        stream: "stream-browserify",
        url: "url",
        util: "util",
        zlib: "browserify-zlib",
      },
    },
    // ...
  };
});
```

**Comment by gioboa** [comment]
Thanks for your report. I solved this issue with #5041 
This update will be available in the next release.

**Comment by niwatoro** [comment]
Thank you @gioboa!
